osx/src/main/java/ch/cyberduck/ui/cocoa/controller/BookmarkController.java

```diff
@@ -216,11 +216,7 @@ public void protocolSelectionChanged(final NSPopUpButton sender) {
                 bookmark.setDefaultPath(selected.getDefaultPath());
             }
             bookmark.setProtocol(selected);
-            final int port = HostnameConfiguratorFactory.get(selected).getPort(bookmark.getHostname());
-            if(port != -1) {
-                // External configuration found
-                bookmark.setPort(port);
-            }
+            bookmark.setPort(HostnameConfiguratorFactory.get(selected).getPort(bookmark.getHostname()));
```

Comment:
Check if `-1` not necessary anymore? Without check the default port is always overridden even if there is no external port set.

Reply:
Changing the protocol will now set the default port regardless of any previous entry.

Reply:
Don't get it. In line 200 the default port is set but overridden here with `-1` unless it's SSH with an alias in `.ssh/config` and custom port.

Reply:
Setting with parameter `-1` will reset to the default port of the current protocol. Will remove the redundant setter.